src/idl_parser.cpp

```diff
@@ -620,6 +620,8 @@ CheckedError Parser::ParseField(StructDef &struct_def) {
   if (struct_def.fixed && !IsScalar(type.base_type) && !IsStruct(type))
     return Error("structs_ may contain only scalar or struct fields");
 
+  bool is_union_vector = false;
```

Comment:
Rather than introduce a new boolean, you can just check below if `typefield` is non-scalar.